langkit/core/workflow.py

```diff
@@ -118,10 +123,40 @@ def get_metric_names(self) -> List[str]:
                 names.extend(metric.names)
         return names
 
-    def run(self, df: pd.DataFrame) -> EvaluationResult:
+    @overload
+    def run(self, data: pd.DataFrame) -> EvaluationResult:
+        """
+        This form is intended for batch evaluation,
+        where the input is a pandas DataFrame.
+        """
+        ...
+
+    @overload
+    def run(self, data: Row) -> EvaluationResult:
+        """
+        This form is intended for single row evaluation,
+        where the input is a dictionary with the keys "prompt" and "response".
+        """
+        ...
+
+    @overload
+    def run(self, data: Dict[str, str]) -> EvaluationResult:
+        """
+        This form doesn't assume the "prompt" and "response" key names.
+        This would be required in cases where the user wants to use different
+        column names, for example "question" and "answer", or "input" and "output".
+        """
+        ...
+
+    def run(self, data: Union[pd.DataFrame, Row, Dict[str, str]]) -> EvaluationResult:
         if not self._initialized:
             self.init()
-
+        if not isinstance(data, pd.DataFrame):
+            if not is_dict_with_strings(data):
+                raise ValueError("Input must be a pandas DataFrame or a dictionary with string keys and string values")
```

Comment:
Row is TypedDict, though, so it would work.
Besides, `isinstance(data, Row)` raises the error `TypeError: TypedDict does not support instance and class checks`

Reply:
ooh I thought it was a data class, my bad